### PR TITLE
Fix issue with connect Envoy choosing incorrect TLS settings.

### PR DESCRIPTION
**Pull request by hashi-derek**
This commit fixes a situation where the API TLS configuration incorrectly influences the GRPC port TLS configuration for XDS. If an agent has TLS enabled on its HTTP endpoints, then it will override the intended gRPC configuration, and envoy will incorrectly get the proper certificates to use.

### CODE REVIEW

File: command/connect/envoy/envoy.go [hunk 1]
```diff
@@ -507,8 +518,15 @@ func (c *cmd) templateArgs() (*BootstrapTplArgs, error) {
 		adminAccessLogPath = DefaultAdminAccessLogPath
 	}
 
+	// Fallback to the old certificate configuration, if none was defined.
+	if xdsAddr.AgentTLS && c.grpcCAFile == "" {
+		c.grpcCAFile = httpCfg.TLSConfig.CAFile
+	}
+	if xdsAddr.AgentTLS && c.grpcCAPath == "" {
+		c.grpcCAPath = httpCfg.TLSConfig.CAPath
+	}
```
Comment (hashi-derek) [inline command/connect/envoy/envoy.go:527]: Is this the correct behavior? It's likely that any customers relying on the old CA cert configuration will break if I don't add in this extra behavior.

Comment (rboyer) [reply command/connect/envoy/envoy.go:527]: so the gist here is "if i am already sure i should use TLS for xds, THEN fall back on the unprefixed flag/envvar place to get the CA stuff from"? If so then that sounds right.

File: .changelog/15466.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+```release-note:bug
+connect: Fix issue where xDS incorrectly uses the HTTPS configuration for connections.
```
Comment (rboyer) [inline .changelog/15466.txt:2]:
```suggestion
cli: Fix issue where `consul connect envoy` incorrectly uses the HTTPS API configuration for xDS connections.
```

**Review by rboyer — APPROVED** [review]

**Comment by rboyer** [comment]
https://github.com/hashicorp/consul/pull/15469

File: .changelog/15466.txt (hunk 2 above)
Comment (Salhazry) [reply .changelog/15466.txt]: Salhazry@gmail.com

Comment (Salhazry) [reply .changelog/15466.txt]: > ```suggestion
> cli: Fix issue where `consul connect envoy` incorrectly uses the HTTPS API configuration for xDS connections.
> ```

Comment (Salhazry) [reply .changelog/15466.txt]: > ```suggestion
> cli: Fix issue where `consul connect envoy` incorrectly uses the HTTPS API configuration for xDS connections.
> ```

Salhazry@gmail.com

**Pull request merged**